### ISSUE TITLE
Curve drawing regression

### ISSUE THREAD

**Issue by mbway** [mozman/ezdxf#473]
I have encountered multiple examples of where a curve that used to be drawn successfully now fails. I narrowed down the changes to: ac923af63c016d0071d2df357ab462ffee4a0eac (add linear Bezier curves as LINE_TO commands #394)

here is the code to create the entities: 
[bug_report.zip](https://github.com/mozman/ezdxf/files/6677315/bug_report.zip)

Before (cdd92b93b218359616d83c4c7e7bc13e85b46469)
![example_1](https://user-images.githubusercontent.com/4923501/122568963-31642e00-d042-11eb-9d0e-26efc0de6d03.png)
![example_2](https://user-images.githubusercontent.com/4923501/122568972-32955b00-d042-11eb-8033-972c87baf5ad.png)
![example_3](https://user-images.githubusercontent.com/4923501/122568986-345f1e80-d042-11eb-8da4-175a8376a1d2.png)

After (ac923af63c016d0071d2df357ab462ffee4a0eac)
![example_1](https://user-images.githubusercontent.com/4923501/122569108-5062c000-d042-11eb-9730-5ae268eab950.png)
![example_2](https://user-images.githubusercontent.com/4923501/122569127-535db080-d042-11eb-94d1-f83f4788317e.png)
![example_3](https://user-images.githubusercontent.com/4923501/122569129-55277400-d042-11eb-9a73-1d2e2eb3d175.png)

**Comment by mozman** [comment]
This is a BIG coordinates and SMALL entity (radius) and `isclose()` issue.

```Python
e = layout.(
    center=(85087205.33123209, 95955965.01566812, 0.0),  # BIG coordinates
    radius=0.24999890076178594,   # SMALL radius
    start_angle=56.39564870647629,
    end_angle=-173.60367671150732,
    dxfattribs={
        'layer': "0",
        'extrusion': (0.0, 0.0, 1.0),
    },
)
```

**Comment by mbway** [comment]
what do you think is the correct solution here then? maybe get rid of the relative tolerance when using isclose for these things or maybe translate the coordinates to the origin? Is isclose even necessary? does AutoCAD use a tolerance or does it use exact equality?

**Comment by mozman** [comment]
This does not fix example3.py which is very extreme:

```
import ezdxf
e = layout.add_circle(
    center=(190233.2851952718, -58473.93874893477, 0.0),  # 190.233  meters
    radius=1.49011612e-08,  # 0.0149 µm 
)
```

The problem is, that the current bezier algorithm crashes for curves where the inner control points are too close to the end points. I think I can't fix the bezier algorithm, it's a text book algorithm and I can't do better than that!

**Comment by mozman** [comment]
Even when I remove the whole "if is_close" statement, cad_viewer.py does not show circle of example_3.dxf.

The debugger shows the correct cubic bezier path creation and the PyQt backend get the correct information to draw the curves, but nothing is to see at the canvas.

The removed if statement is here: https://github.com/mozman/ezdxf/blob/cac71451e731fe3c3b149b88a4b4b7eb26aa88f9/src/ezdxf/path/tools.py#L621

**Comment by mbway** [comment]
personally, I don't mind that much if the Qt backend ignores it because I'm using the drawing framework as a file format conversion. The main issue for me is that the occurrence of straight lines where I don't expect them is causing issues. I could just entirely skip entities where this kind of thing occurs but that could mask other issues.

If I remove the checks introduced in ac923af63c016d0071d2df357ab462ffee4a0eac and run the example code from this issue or draw the provided dxf https://github.com/mozman/ezdxf/issues/394 (which I assume you're referring to when talking about crashing the Bezier algorithm) it still doesn't crash for me. Are there other examples where these guards prevent crashes?

**Comment by mozman** [comment]
The check is not problem, because the result in cad_viewer.py is the same without this checks (no circle at all), there is something else going on.

I also tested `math.isclose()`: `math.isclose(190000.00000001, 190000.00000002, rel_tol=1e-14)` to get `False`

**Comment by mozman** [comment]
Status update ezdxf v0.16.6b0 - nothing changed:
`example_1.dxf` and `example_2.dxf`: correct result for the Matplotlib and the PyQt5 backend
`example_3.dxf`: only the Matplotlib backend shows the correct result, the PyQt5 backend still shows nothing

**Comment by mbway** [comment]
PyQt not showing an entity could be because the Qt canvas uses integer coordinates and the entity has a size of less than 1 unit? for a project of mine where I render to a QGraphicsView I scale everything by 10,000 (arbitrary choice) to overcome this limitation

**Comment by mozman** [comment]
Scaling the content fixed `example_3.dxf`.

**Comment by mozman** [comment]
Added a `--scale` argument to `ezdxf view` command, but this does no solve the issue.

Unscaled paths points:
![unscaled_path](https://user-images.githubusercontent.com/4230966/126889834-f4448124-4bca-4e90-957e-afea0cc43714.png)

Scaled path points (--scale=1e12)
![scaled_path](https://user-images.githubusercontent.com/4230966/126889848-141c6baa-ab10-4b1c-b610-82d3b59c7ac4.png)

The PyQt-Backend does not show the ARC at any factor: 1000, 1e6, 1e9 or 1e12.

Setting the radius of the ARC to 1 in BricsCAD solves the issue.

I have no further ideas, but at least I removed the double rendering of the model space in the `view` command (1st in `set_document()` and 2nd in `draw_layout()`).

I will keep the scaling argument, maybe it is useful for something else :).
